### ISSUE TITLE
Dealing with unbalanced datasets

### ISSUE THREAD

**Issue by supvato** [open-edge-platform/geti_v2#1427]
### 🧩 Issue type

Documentation 📚

### 📄 Description

It is not clear from the documentation if the software compensates for unbalanced datasets or if there is a way to do this manually. Could you please clarify.

### 📦 Geti version

2.12.1

**Comment by jgespino** [comment]
@supvato When training a model, under advance options there are setting for data augmentation to enhance the dataset. Currently it's only available for classification models. Take a look at this [section](https://docs.geti.intel.com/docs/user-guide/geti-fundamentals/model-training-and-optimization/?_highlight=data&_highlight=augmentation#train-model) of the documentation for more information.

**Comment by supvato** [comment]
@jgespino, thank you for you reply. I looked into the suggested documentation. I doesn't look like data augmentation will help with an unbalanced (aka imbalanced) dataset. Augmentation won't change the number of samples in each class, it only modifies the images to make the model more robust. 
Pytorch has a few methods to deal with imbalanced datasets. For example, by using `pos_weight` in `BCEWithLogitsLoss()`. Other methods consist of sampling the minority class more often to make up for the imbalance. However, I don't see anything similar in the documentation for Geti.

**Comment by leoll2** [comment]
Hi @supvato , Geti does not implement resampling methods like the one that you mention (oversampling the minority class). On the other hand, it uses some mechanisms that have an effect in mitigating (or not worsening) situations of data imbalance, for example:

- Data augmentations: even if the proportion of classes does not change, training the model with a larger number (in absolute terms) of samples of the rare class makes it more robust for that class.
- Loss functions: the training backend exploits loss functions that penalize misclassification of infrequent classes, or put more weight on hard-to-classify examples, such as focal loss
- Subset splitting: items are assigned to train/validation/test subsets trying to keep a balanced distribution of the classes among the three. This prevents undesirable situations where you have very few samples of a certain class and, due to random assignment, most of them (if not all) end up in the same subset.
- Evaluation metrics: certain metrics like F1 score are suitable for evaluation on imbalanced datasets

Finally, I'd like to point out that Geti also supports the _anomaly detection_ task, whose algorithms address class imbalance problems by design ("anomalous" samples are rare/hard to collect in many real world scenarios). Anomaly detection may or may not be a good fit depending on your usecase, but it's still an option to consider.
